driver/utils.hpp

```diff
@@ -196,7 +195,23 @@ void check_solution( int n, vdist_int * vertexDist, ptr_t * colptr, ind_t * rowi
   }
 }
 
-
+size_t get_gpu_alloc_size(const std::string& argstring) {
+    size_t size; int suffix_start;
+    
+    suffix_start = argstring.find_first_not_of("0123456790");
+    
+    std::string alloc_size_str = argstring.substr(0, suffix_start);
+    std::string alloc_units = argstring.substr(suffix_start);
+
+    size = std::stoul(alloc_size_str);
+    
+    if (alloc_units=="MiB")
+        size *= (1<<20);
+    else if (alloc_units=="GiB")
+        size *= (1<<30);
+    
+    return size;
```

Comment:
Thanks for adding this nice usability improvement!

However I'm concerned the implementation above doesn't contain any "guardrails" to validate the user input is correctly recognized, leading to potential nonsense. In particular, I'm concerned about silent misbehavior if the user types one of the following:

```
run_sympack2D -gpu_mem 2 GiB
run_sympack2D -gpu_mem 2.5GB
run_sympack2D -gpu_mem 2GB
run_sympack2D -gpu_mem 2G
run_sympack2D -gpu_mem 2gib
run_sympack2D -gpu_mem 2
```

These all technically violate the documentation in README, but you can understand how a user might easily do these. I suggest the following enhancements:

1. Append all of `options["-gpu_mem"]` to include trailing options
2. Parse the number as floating point
3. Interpret any unit that starts with [gGmM] as GiB or MiB respectively and ignore subsequent characters
4. Issue an error if the first unit character is anything else
5. Issue an error if the parsed gpu_mem segment value is less than 1 MiB (which is effectively nonsense, and means we misunderstood the user)

If you're feeling ambitious, you could even factor this parsing function, add a [kK] suffix for KiB, and use it to also parse memory size value provided in the `-gpu_blk` and `-*_limit` arguments (although I think those are far less likely to be tweaked in practice by end users).

Reply:
Resolved as suggested in c62c5ad